genvm/vm_test.go

```diff
@@ -2046,6 +2046,94 @@ func FuzzParse(f *testing.F) {
 	})
 }
 
+func BenchmarkTransactions(b *testing.B) {
+	bench := func(b *testing.B, tt *tester, raw types.RawTx) {
+		b.Log("tx size", len(raw.Raw))
+		lid := types.GetEffectiveGenesis().Add(2)
+		for i := 0; i < b.N; i++ {
+			b.StartTimer()
+			ineffective, txs, err := tt.Apply(ApplyContext{Layer: lid}, []types.Transaction{{RawTx: raw}}, nil)
+			b.StopTimer()
+			require.NoError(b, err)
+			require.Empty(b, ineffective)
+			require.Equal(b, types.TransactionSuccess, txs[0].Status)
+			require.NoError(b, tt.Revert(lid.Sub(1)))
+		}
+	}
+
+	// benchmarks below will have overhead beside the transaction itself.
+	// they are useful mainly to collect execution profiles and make estimations based on them.
+	b.Run("singlesig/selfspawn", func(b *testing.B) {
```

Comment:
i am curious why you omitted spawnTx in the benchmark.

Reply:
there is a spawnvault, i thought it is enough to understand relative difference, can add more